testbed/correctness/utils.go

```diff
@@ -0,0 +1,154 @@
+package correctness
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+func CreateConfigYaml(
+	sender testbed.DataSender,
+	receiver testbed.DataReceiver,
+	processors map[string]string,
+	pipelineType string,
+) string {
+
+	// Prepare extra processor config section and comma-separated list of extra processor
+	// names to use in corresponding "processors" settings.
+	processorsSections := ""
+	processorsList := ""
+	if len(processors) > 0 {
+		first := true
+		for name, cfg := range processors {
+			processorsSections += cfg + "\n"
+			if !first {
+				processorsList += ","
+			}
+			processorsList += name
+			first = false
+		}
+	}
+
+	format := `
+receivers:%v
+exporters:%v
+processors:
+  %s
+
+extensions:
+
+service:
+  extensions:
+  pipelines:
+    %s:
+      receivers: [%v]
+      processors: [%s]
+      exporters: [%v]
+`
+
+	return fmt.Sprintf(
+		format,
+		sender.GenConfigYAMLStr(),
+		receiver.GenConfigYAMLStr(),
+		processorsSections,
+		pipelineType,
+		sender.ProtocolName(),
+		processorsList,
+		receiver.ProtocolName(),
+	)
+}
+
+type PipelineDef struct {
+	Receiver     string
+	Exporter     string
+	TestName     string
+	DataSender   testbed.DataSender
+	DataReceiver testbed.DataReceiver
+	ResourceSpec testbed.ResourceSpec
+}
+
+func LoadPictOutputPipelineDefs(fileName string) ([]PipelineDef, error) {
```

Comment:
Comments please for exported types and funcs :) Paying the cost of documentation now that you expose them :)

Reply:
No problem, done

Reply:
Thanks